src/trusted/service_runtime/nacl_syscall_common.c

```diff
@@ -4330,3 +4320,104 @@ int32_t NaClSysSelect (struct NaClAppThread *natp, int nfds, fd_set * readfds,
   
   return retval;
 }
+
+int32_t NaClSysSigaction(struct NaClAppThread *natp, int32_t sig, const struct nacl_abi_sigaction *act, struct nacl_abi_sigaction *oact) {
+  int32_t ret;
+  struct NaClApp *nap = natp->nap;
+  const struct nacl_abi_sigaction *sysact = NULL;
+  struct nacl_abi_sigaction *sysoact = NULL;
+
+  if (act) {
+    sysact = (const struct nacl_abi_sigaction*) NaClUserToSysAddrRangeProt(nap, (uintptr_t) act, sizeof(struct nacl_abi_sigaction), NACL_ABI_PROT_READ);
+
+    if ((void*) kNaClBadAddress == sysact) {
+        NaClLog(2, "NaClSysSigaction could not translate act, returning %d\n", -NACL_ABI_EFAULT);
+        return -NACL_ABI_EFAULT;
+    }
+  }
+
+  if (oact) {
+    sysoact = (struct nacl_abi_sigaction*) NaClUserToSysAddrRangeProt(nap, (uintptr_t) oact, sizeof(struct nacl_abi_sigaction), NACL_ABI_PROT_WRITE);
```

Comment:
I think I'm confident that we don't need a `WRITE` here, since we won't change the old action in-place, we only need `NACL_ABI_PROT_READ`.

Reply:
No, old action gets populated by rustposix, it needs to be writeable.